lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -393,7 +392,7 @@ private void WaitForTaskRegistration(IList<string> identifiers, CancellationToke
 
                 var leftOver = string.Join(",", identifiers.Where(e => !foundList.Contains(e)));
                 Logger.Log(Level.Error, "For node {0}, cannot find registered parent/children: {1}.", _selfId, leftOver);
-                Exceptions.Throw(new RemotingException("Failed to find parent/children nodes in operator topology for node: " + _selfId), Logger);
+                Exceptions.Throw(new InvalidOperationException("Failed to find parent/children nodes in operator topology for node: " + _selfId), Logger);
```

Comment:
We use RemoteException for this case to represent some connection issue. This could happen when one of the node fails that makes parent/children cannot connect to it. We categorize this RemoteException as CommunicationException that is recoverable. 
InvalidOperationException is a real error that is not recoverable. 

We can use either TcpClientConnectionException or SocketException , or create our own RemoteException to represent this case.